apps/test/unit/weblab/WebLabTest.js

```diff
@@ -12,6 +28,67 @@ describe('WebLab', () => {
     weblab = new WebLab();
   });
 
+  describe('init', () => {
+    let config;
+    beforeEach(() => {
+      stubRedux();
+      stubStudioApp();
+      weblab.studioApp_ = studioApp();
+      registerReducers(commonReducers);
+      registerReducers(reducers);
+      config = {
+        skin: {},
+        level: {}
+      };
+      sinon.stub(ReactDOM, 'render');
+      sinon.stub(getStore(), 'dispatch');
+    });
+
+    afterEach(() => {
+      restoreRedux();
+      restoreStudioApp();
+      ReactDOM.render.restore();
+    });
+
+    it('throws an error if studio app doesnt exist', () => {
+      weblab.studioApp_ = null;
+      expect(weblab.init).to.throw(Error);
+    });
+
+    it('dispatches changeMaxProjectCapacity', () => {
+      weblab.init(config);
+      expect(getStore().dispatch).to.have.been.calledWith(
+        changeMaxProjectCapacity(20971520)
+      );
+      expect(ReactDOM.render).to.have.been.calledOnce;
+    });
+
+    it('does not set startSources if there are none', () => {
+      config.level.startSources = '';
+      weblab.init(config);
+      expect(weblab.startSources).to.be.undefined;
+    });
+
+    it('does not set startSources if it is given invalid JSON', () => {
+      config.level.startSources = '{:';
+      expect(() => weblab.init(config)).to.throw(Error);
+      expect(weblab.startSources).to.be.undefined;
+    });
+
+    it('sets startSources if given valid JSON', () => {
+      const validJSON = {value: 'test'};
+      config.level.startSources = JSON.stringify(validJSON);
+      weblab.init(config);
+      expect(weblab.startSources.value).to.equal('test');
+    });
+
+    it('calls the filesApi to get files', () => {
+      sinon.spy(filesApi, 'getFiles');
+      weblab.init(config);
+      expect(filesApi.getFiles).to.have.been.calledOnce;
+    });
```

Comment:
this test is actually testing `loadFileEntries`, which has enough logic that it should probably have its own set of tests (e.g., it's own `describe` block with multiple test cases)

Reply:
Ok, I'll take that out for now and make loadFileEntries its own set of tests in the next pass